### ISSUE TITLE
"Tools" menu items remain expanded after navigating to a different page

### ISSUE THREAD

**Issue by aneeshd16** [Automattic/wp-calypso#49110]
#### Steps to reproduce the behavior

1. Go to My Home
2. Hover on Tools
3. Click on Marketing
4. Click on My Home

I could not reproduce this with any other menu item, but an investigation into the cause might give hints on whether other menu items are affected.

#### What I expected to happen
The Tools menu should not be shown since I am not hovering on it.

#### What actually happened
The expanded (on hover) state of the Tools menu was shown.

### Browser / OS version
Chrome Version 87.0.4280.141 (Official Build) (x86_64), macOS Catalina 10.15.6

### Does this happen on simple or atomic sites or both?
Happens on both simple and atomic.
### Is there any console output or error text?
No
### Level of impact (Does it block purchases? Does it affect more than just one site?)
Low
#### Screenshot / Video: If applicable, add screenshots to help explain your problem.
![image](https://user-images.githubusercontent.com/5436027/105341040-a6067e00-5c04-11eb-9da8-af429ade287a.png)
![Kapture 2021-01-21 at 16 23 04](https://user-images.githubusercontent.com/5436027/105341339-f978cc00-5c04-11eb-81f6-62fbc882d5b2.gif)

**Comment by deancroyal** [comment]
I was able to reproduce this, but using different menu items.

**Steps to reproduce:**

1. Go to My Home.
2. Mouse over Appearance in the menu and click on Themes within the submenu.
3. Click on Media.

**What I expected:** To be taken to Media, and only Media be highlighted in the menu.

**What happened:** I was taken to Media and Media was highlighted in the menu, however, the Appearance menu item was also shown expanded.

**Additional Finding:** After performing steps 1 & 2 from above, I found I can consecutively click on Media, then Posts, then Pages, then Comments, and the expanded Appearance menu item continued to display throughout. But, if I clicked on the Links menu item, the expanded Appearance menu item did not appear. (I demonstrated this in the video below.)

**Browser / OS version**
Version 87.0.4280.141 (Official Build) (x86_64) / macOS Big Sur 11.1 (20C69)

**Does this happen on simple or atomic sites or both?**
I was only able to test on a simple site, which is where I found the issue.

**Is there any console output or error text?**
No

**Level of impact** (Does it block purchases? Does it affect more than just one site?)
Low

Screenshot / Video: If applicable, add screenshots to help explain your problem.

https://user-images.githubusercontent.com/13387217/105375214-623c5600-5bbd-11eb-8269-83844745264c.mov
